snapcraft/remote/errors.py

```diff
@@ -50,3 +50,22 @@ def __init__(self, message: str) -> None:
         details = message
 
         super().__init__(brief=brief, details=details)
+
+
+class RemoteBuildTimeoutError(RemoteBuildError):
+    """Remote-build timed out."""
+
+    def __init__(self) -> None:
+        brief = "Remote build timed out."
```

Comment:
I agree this error isn't very helpful.  However, the user shouldn't be too surprised to see it, since the only way you can get this error is by passing the command line argument `--launchpad-timeout=<seconds>`.

On a related note, I need to ask @sergiusens on why `--launchpad-timeout` was dropped in snapcraft 7.  I can't tell if it was intentional or not.

Reply:
Yeah, this is more a vague UX-related thought, not a change request because as-is there isn't any possible change.